sophiread/FastSophiread/src/tpx3_fast.cpp

```diff
@@ -298,6 +375,35 @@ void update_tdc_timestamp(const char *char_array,
   tdc_timestamp = tdc_timestamp | mytdc;
 }
 
+/**
+ * @brief Update the tdc_timestamp without relying on GDC reference
+ *
+ * @param[in] char_array: the char array of the data packet
+ * @param[out] tdc_timestamp: the target to update
+ */
```

Comment:
This function is dealing with the TDC roll over issue, did you get it from Vlad method or Zach's implementation?
I don't recall seeing this in either implementation.